lib/LANraragi/Utils/Minion.pm

```diff
@@ -140,6 +141,100 @@ sub add_tasks {
         }
     );
 
+    $minion->add_task(
+        find_duplicates => sub {
+            my ( $job,      @args )  = @_;
+            my ( $threshold ) = @args;
+
+            my $logger = get_logger( "Minion", "minion" );
+            my $redis  = LANraragi::Model::Config->get_redis;
+            my @keys   = $redis->keys('????????????????????????????????????????');
+            $redis->quit();
+
+            $logger->info("Starting find duplicate job (threshold = $threshold)");
+
+            my $numCpus = Sys::CpuAffinity::getNumCpus();
+            my $pl      = Parallel::Loops->new($numCpus);
+
+            $logger->debug("Number of available cores for processing: $numCpus");
+
+            # Gather thumbhashes
+            my %thumbhashes;
+            foreach my $id (@keys) {
+                my $thumbhash = $redis->hget( $id, "thumbhash" );
+                $thumbhashes{$id} = $thumbhash if $thumbhash;
+            }
+
+            # Prepare to track visited nodes
+            my %visited :shared;
+            my @ids = keys %thumbhashes; # List of IDs to check
+
+            # Share the %visited hash across threads
+            $pl->share( \%visited );
+
+            my @sections = split_workload_by_cpu( $numCpus, @ids );
+            eval {
+                $pl->foreach(
+                    \@sections,
+                    sub {
+                        my $redis = LANraragi::Model::Config->get_redis;
+
+                        foreach my $id (@$_) {
+                            # Skip if this ID has already been processed in another thread
+                            next if $visited{$id};
```

Comment:
Since you've used `split_workload_by_cpu`, normally each process/thread should have its own unique set of IDs.  

I'm not sure you need visited as a result?